backend/coins/eth/feetarget.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eth
+
+import (
+	"math/big"
+	"strings"
+
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/accounts"
+)
+
+// feeTarget contains the gas price for a specific fee target.
+type feeTarget struct {
+	// Code is the identifier for the UI.
+	code accounts.FeeTargetCode
+	// gasPrice is the estimated gas price to be used in the fee calculation, in Wei.
+	gasPrice *big.Int
+}
+
+// Code returns the btc fee target.
+func (f *feeTarget) Code() accounts.FeeTargetCode {
+	return f.code
+}
+
+// FormattedFeeRate returns a string showing the fee rate.
+func (f *feeTarget) FormattedFeeRate() string {
+	if f.gasPrice == nil {
+		return ""
+	}
+	factor := big.NewInt(1e9)
+	return strings.TrimRight(strings.TrimRight(
+		new(big.Rat).SetFrac(f.gasPrice, factor).FloatString(9),
+		"0"), ".") + " Gwei"
```

Comment:
spent at least few minutes trying to match parenthesis. is the idea to remove trailing zeros? this should be equivalent but easier to read imho:

```go
s := new(big.Rat).SetFrac(f.gasPrice, factor).FloatString(9)
return strings.TrimRight(s, ".0") + " Gwei"
```

Reply:
Your code turns `10` into `1`.

We use this type of formatting in a few places in the app and also in the firmware: https://github.com/digitalbitbox/bitbox02-firmware/blob/master/src/rust/util/src/decimal.rs#L32

I'll try to add unit tests.

Reply:
Ah it doesn't cause FloatString adds decimal places. Hmm

Reply:
> Your code turns 10 into 1.

sounds like even more reason to comment the code and make it more readable.

Reply:
I added unit tests, your code doesn't work for e.g. `21e10` which becomes `210 Gwei` - source code strips beyond the decimal place to `21 Gwei`.

Reply:
those 3 lines are still unreadable.

Reply:
improved

Reply:
much nicer. thanks! :+1: